src/components/LeaderBoard.jsx

```diff
@@ -0,0 +1,249 @@
+import React, { Component, Fragment } from 'react';
+import Avatar from 'react-avatar';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Link } from 'react-router-dom';
+
+import config from 'configuration';
+import Loader from './Loader';
+import { getUserName, getUserAvatar, convertEthHelper } from '../lib/helpers';
+import Donation from '../models/Donation';
+
+/**
+ * Shows a table of aggregateDonations for a given type (dac, campaign, milestone)
+ */
+
+class LeaderBoardItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showDetails: false,
+      itemType: null, // Delegated ro Directly donated
+      hasHistory: this.props.d.status === Donation.PAID, // PAID donations always has history
+    };
+
+    this.toggleDetail = this.toggleDetail.bind(this);
+    this.setItemType = this.setItemType.bind(this);
+    this.setItemHasHistory = this.setItemHasHistory.bind(this);
+  }
+
+  setItemType(type) {
+    this.setState({
+      itemType: type,
+    });
+  }
+
+  setItemHasHistory(hasHistory) {
+    this.setState({
+      hasHistory,
+    });
+  }
+
+  toggleDetail() {
+    this.setState(prevState => ({
+      ...prevState,
+      showDetails: !prevState.showDetails,
+    }));
+  }
+
+  render() {
+    const { d, rank, useAmountRemaining } = this.props;
+    const { donations, totalAmount, giver } = d;
+    return (
+      <Fragment>
+        <tr key={d._id}>
+          <td>
+            <button type="button" className="btn btn-info btn-sm" onClick={this.toggleDetail}>
+              <i className={this.state.showDetails ? 'fa fa-minus' : 'fa fa-plus'} />
+            </button>
+          </td>
+          <td className="font-weight-bold">{rank}</td>
+          <td className="td-user">
+            {giver && (
+              <Link to={`/profile/${giver.address}`}>
+                <Avatar size={30} src={getUserAvatar(giver)} round />
+                <span> {getUserName(giver)}</span>
+              </Link>
+            )}
+          </td>
+          <td className="td-donations-amount font-weight-bold">${totalAmount}</td>
+          <td />
+          <td className="" />
+        </tr>
+        {donations.map(dd => {
+          const donation = new Donation(dd);
```

Comment:
- Is it necessary to create Donation object?
- Please create a separate component for showing donation rows, it's necessary since we need to get donation type (direct/delegate) by using another module and passing it some setter function